gatsby/content/blog/2022/10/2022-10-18-faster-joins.mdx

```diff
@@ -0,0 +1,43 @@
+
+Firstly, and most importantly: **we do not yet recommend enabling faster joins on important production systems**. We've done quite a lot of testing, but there are some fundamental changes to Synapse, and by their nature they are hard to test rigorously. We're pretty confident it won't eat your entire database, but stuck rooms, stuck notifications, and stuck clients, are all entirely within the realms of possibility. We'd recommend only enabling faster joins on your server if you are comfortable looking at the logs, and using the Synapse Admin API or flushing client caches to clean things up if they go wrong.
+
+Secondly: there's still some [outstanding work](https://github.com/matrix-org/synapse/issues/12994) required to support faster joins on Synapse deployments using worker processes. At present, if you try to enable them on a worker-based deployment, Synapse will refuse to start.
+
+Next: the faster-joins process requires some support from the resident server, and, since [those changes](https://github.com/matrix-org/matrix-spec-proposals/pull/3706) are not yet in the stable matrix spec, it must be explicitly enabled. We have enabled this support on the `matrix.org` homeserver, but it generally won't work elsewhere. In other words: for now, you'll only see the benefit when you join a room via a `#<name>:matrix.org` alias (or via the `matrix.org` room directory).
+
+Finally, there are still quite a few things that don't work properly yet. We're tracking the list of things we need to fix as a [milestone](https://github.com/matrix-org/synapse/milestone/10) on the Synapse project, but to name a few:
+
+ * You can't yet send messages during the "resynchronisation" phase ([synapse#12997](https://github.com/matrix-org/synapse/issues/12997)). Currently, you'll just get a spinner.
+ * Similarly, attempts to leave the room ([synapse#12802](https://github.com/matrix-org/synapse/issues/12802)) will block until the resync completes.
+ * Clients which don't support [lazy-loading room members](https://spec.matrix.org/v1.4/client-server-api/#lazy-loading-room-members) will *block* (ie, they won't receive any new events at all) during the resynchronisation process ([synapse#12989](https://github.com/matrix-org/synapse/issues/12989)). Most popular clients, including Element, Hydrogen, Fractal and FluffyChat *do* support lazy-loading, but a few (including Nheko) do not.
+
+## Turning it on
+
+With all that said: we are very excited for people to start trying it and give us feedback. If you administer a Synapse server and are as excited as we are, all you need to do to start testing faster joins is to add this to your `homeserver.yaml` (and then restart Synapse):
+
+```yaml
+experimental_features:
+  faster_joins: True
+```
+
+With that, any future remote room joins (or at least those going via `matrix.org`) will use the new faster joins algorithm. Please let us know how you get on, and file Github issues for any problems you might have!
```

Comment:
I wonder if we're going to need a checkbox on the issue template for "are you using faster joins?"